board/views.py

```diff
@@ -41,7 +46,208 @@ def get(self, request):
+        except ValueError:
+            return JsonResponse({'message' : 'DECODE_ERROR'}, status = 400)
+        except Comment.DoesNotExist:
+            return JsonResponse({'message' : 'NO_EXIST_DATA'}, status = 500)        
+        except :
+            return JsonResponse({'message' : 'NO_COMMENT_EXIST'}, status = 500)
+
+    ## 2. 댓글 작성(POST)
+    #@login_decorator
+    def post(self, request):
+        try:
+            data        = json.loads(request.body)
+            board_id    = data['board_id']
+            member_id   = data['member_id']
+            content     = data['content']
+
+            if content == "":
+                return JsonResponse({'message' : 'COMMENT_REQUIRED'}, status=400)
+            else:
+                Comment.objects.create(
+                    content         = content,
+                    created_at      = datetime.datetime.now(),
+                    board_id        = board_id,
+                    writer_id       = member_id
+                )
+            return JsonResponse({'message' : 'SUCCESS'}, status = 201)
+        except KeyError:
+            return JsonResponse({'message' : 'KEY_ERROR'}, status = 400)          
+        except :
+            return JsonResponse({'message' : 'NO_COMMENT_EXIST'}, status = 500) 
+
+
+# 3. 대댓글 작성
+class AddSelfCommentView(View):
+    #@login_decorator
+    def post(self, request):
+        try:
+            data        = json.loads(request.body)
+            board_id    = data['board_id']
+            member_id   = data['member_id']
+            comment_id  = data['comment_id']
+            content     = data['content']
+
+            if board_id and member_id:
+                if content == "":
+                    return JsonResponse({'message' : 'COMMENT_REQUIRED'}, status=400)
+                elif Comment.objects.filter(id=comment_id).exists():
+                    Comment.objects.create(
+                        content         = content,
+                        created_at      = datetime.datetime.now(),
+                        board_id        = board_id,
+                        self_comment_id = comment_id,
+                        writer_id       = member_id
+                    )
+            return JsonResponse({'message' : 'SUCCESS'}, status = 201)
+        except KeyError:
+            return JsonResponse({'message' : 'KEY_ERROR'}, status = 400)
+        except :
+            return JsonResponse({'message' : 'NO_COMMENT_EXIST'}, status = 500)
+
+
+# 4. 게시물 좋아요
+class LikeBoardView(View):
+    #@login_decorator
+    def post(self, request):
+        try:
+            data        = json.loads(request.body)
+            board_id    = data['board_id']
+            member_id   = data['member_id']
+
+            if BoardLike.objects.filter(board_id=board_id, member_id=member_id).exists():
+                if BoardLike.objects.filter(board_id=board_id, member_id=member_id)[0].is_like == 1:
```

Comment:
```suggestion
                if BoardLike.objects.filter(board_id=board_id, member_id=member_id)[0].is_like == 1:
```
하나의 게시물에 유저가 좋아요를 남길경우는 한번이므로 get().is_like==1이 낫지 않을까요?

Reply:
확인했습니다!